src/autolink.c

```diff
@@ -139,6 +139,55 @@ autolink_delim(uint8_t *data, size_t link_end, size_t max_rewind, size_t size)
 	return link_end;
 }
 
+/*
+ * Checks that `prefix_char` occurs on a word boundary just before `data`,
+ * where `data` points to the character to search to the left of, and a word boundary
+ * is (currently) a whitespace character, punctuation, or the start of the string.
+ * Returns the length of the prefix.
+ */
+static int
+check_reddit_autolink_prefix(
+	const uint8_t* data,
+	size_t max_rewind,
+	size_t max_lookbehind,
+	size_t size,
+	char prefix_char
+	)
+{
+	/* Make sure this `/` is part of `/?r/` */
+	if (size < 2 || max_rewind < 1 || data[-1] != prefix_char)
+		return 0;
+
+	/* Not at the start of the buffer, no inlines to the immediate left of the `prefix_char` */
+	if (max_rewind > 1) {
+		const char boundary = data[-2];
+		if (boundary == '/')
+			return 2;
+		/**
+		 * Here's where our lack of unicode-awareness bites us. We can't correctly
+		 * match punctuation / whitespace characters for the boundary, because we
+		 * might reject valid cases like "。r/example" (note the fullwidth period.)
+		 *
+		 * A better implementation might try to rewind over bytes with the 8th bit set, try
+		 * to decode them to a valid codepoint, then do a unicode-aware check on the codepoint.
+		 */
+		else if (ispunct(boundary) || isspace(boundary))
```

Comment:
Additional note on this line. If we *wanted* to support `\r/foo` not linking, this would be where it would be implemented, right? Essentially check to see if `boundary == '\\'` and return 0 if so?

If so, and it's that simple, then I think we probably should support it just so that it matches the behavior of `\/r/foo`. I really don't feel too strongly if it's more complex than that, though.